shared/src/main/scala/mlscript/ConstraintSolver.scala

```diff
@@ -203,12 +203,13 @@ class ConstraintSolver extends NormalForms { self: Typer =>
                     case _ => reportError
                   }
               }
-            case (LhsRefined(N, ts, r), RhsBases(pts, N | S(L(_: FunctionType | _: TupleType)))) =>
-              reportError
+            case (LhsRefined(N, ts, r), RhsBases(pts, N | S(L(_: FunctionType | _: ArrayBase)))) => reportError
```

Comment:
We try to keep lines below 100 columns.

```suggestion
            case (LhsRefined(N, ts, r), RhsBases(pts, N | S(L(_: FunctionType | _: ArrayBase)))) =>
              reportError
```